diff --git a/src/GNUmakefile b/src/GNUmakefile
index c4f8fd2..7264466 100644
--- a/src/GNUmakefile
+++ b/src/GNUmakefile
@@ -17,6 +17,8 @@ OBJDUMP  ?= $(CROSS_COMPILE)objdump
 
 CPPFLAGS += -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -I. -I$(VPATH)
 CFLAGS   += -Wall -Wextra -O2
+PROOT_MAIN_CFLAGS ?=
+PROOT_CLI_CFLAGS ?= $(PROOT_MAIN_CFLAGS)
 LDFLAGS  += -ltalloc -Wl,-z,noexecstack
 ifdef PROOT_WITH_LIBANDROID_SHMEM
   CFLAGS += -DWITH_LIBANDROID_SHMEM
@@ -180,4 +182,6 @@ ROSEGMENT := $(shell printf 'int main(void){return 0;}\n' | $(LD) -xc - -Wl,--r
 COMPILE = $($(quiet)CC) $(CPPFLAGS) $(CFLAGS) -MD -c $(SRC)$< -o $@
+COMPILE_MAIN = $(COMPILE) $(PROOT_MAIN_CFLAGS)
+COMPILE_CLI = $(COMPILE) $(PROOT_CLI_CFLAGS)
 LINK    = $($(quiet)LD) -o $@ $^ $(LDFLAGS)
 
 OBJIFY = $($(quiet)GEN)									\
@@ -199,7 +203,11 @@ $(USE_BUILD_H): build.h
 
+cli/cli.o: cli/cli.c
+	@mkdir -p $(dir $@)
+	$(COMPILE_CLI)
+
 %.o: %.c
 	@mkdir -p $(dir $@)
-	$(COMPILE)
+	$(COMPILE_MAIN)
 
 cli/%-licenses.o: licenses cli/cli.o
 	$(OBJIFY)
